src/main/java/org/folio/dao/DataImportQueueItemDao.java

```diff
@@ -15,24 +17,26 @@ public interface DataImportQueueItemDao {
   Future<DataImportQueueItemCollection> getAllQueueItems();
 
   /**
-   * Get all {@link DataImportQueueItem} in database where processing=false
+   * Get all {@link DataImportQueueItem} in database where processing=false using the specified {@code connection}
```

Comment:
why is a connection needed?

Reply:
connection is needed to execute the method/query in transaction initiated within specified connection. getAllQueueItemsAndProcessAtomic method is intended to ensure that only one worker can process certain queue item atomically, and this is done by obtaining exclusive lock on queue items table. That's why the search in the queue items table should be executed within transaction that acquired the lock